python-frontend/src/main/java/org/sonar/python/types/TypeInference.java

```diff
@@ -94,39 +96,55 @@ private static Set<Symbol> getTrackedVars(Set<Symbol> localVariables, Set<Name>
       boolean hasMissingBindingUsage = variable.usages().stream()
         .filter(Usage::isBindingUsage)
         .anyMatch(u -> !assignedNames.contains(u.tree()));
-      if (!hasMissingBindingUsage) {
+      boolean isGlobal = variable.usages().stream().anyMatch(v -> v.kind().equals(Usage.Kind.GLOBAL_DECLARATION));
+      if (!hasMissingBindingUsage && !isGlobal) {
         trackedVars.add(variable);
       }
     }
     return trackedVars;
   }
 
-  private static void inferTypesAndMemberAccessSymbols(FunctionDef functionDef, PythonFile pythonFile) {
+  private static void inferTypesAndMemberAccessSymbolsForFile(FileInput fileInput, PythonFile pythonFile) {
```

Comment:
as discussed we decided to extend the common logic between `inferTypesAndMemberAccessSymbolsForFile` and `inferTypesAndMemberAccessSymbolsForFunction` to another method